models/comment.js

```diff
@@ -0,0 +1,19 @@
+const mongoose = require('mongoose')
+
+const commentSchema = mongoose.Schema({
+	_id: mongoose.Schema.Types.ObjectId,
+    email: {
```

Comment:
You need to provide ref of User model to populate this in a query. Please do add that and also see if we can populate User model with this user email or not.

Reply:
I am going to add support for `ObjectId` in `User` collection as it is missing, and add ref in this collection through `User._id`. So the comment will only get created if valid user exist. That is going to be faster as it will be indexed query.
And also in token generation, I am going to generate token through `User._id`, and not through `email`.

Reply:
Adding explicitly User._id is not necessary. Mongodb automatically adds _id even if not specified in schema

Reply:
Okay, and how about token generation through `_id`?

Reply:
What do you mean by token generation? Do you mean jwt tokens?

Reply:
Yes that's what. As of now we generate token by encrypting `email`. How about doing so by `_id`

Reply:
Yes we can do that but the reason to use email in tokens is that we do not need to find user with that `_id` in API's that require just the emails in future.

Reply:
How about doing both? We can encrypt `"<_id>::<email>"`.

Reply:
It's better to keep payload as minimal as possible. If you think that adding this will help in optimising the APIs then sure we can do that.